### DISCUSSION TITLE
Add Onclick event on Table row

### DISCUSSION THREAD

**Discussion by phangs** [Budibase/budibase#1598] [Ideas]
It would be easier for us non-coders to be able to able to add an onclick even to a table row, this might trigger opening the record in a new window or even a modal to display the whole record which is impossible to be displayed in a table (if the record has too many columns). Event can be a javascript function that we can write which use the id of the selected row for reference.

Having an onclick event will give the table component more usability in the future.

**Comment by kevmodrome** [comment]
That's certainly something to explore. In the meantime you can add a button inside of the table that you can hook up to an action. If you want to see an example of how this looks - create an internal table and then go to the design section.

**Reply by phangs** [reply]
thank you very much!

What I'm actually getting at is the table gets the data from the external data source, when a user clicks on a row/record, a new window/modal will popup with the record fields are mapped, then the user may modify the information then a button can be use to trigger an http post to the external API source to update the record.

**Reply by joebudi** [reply]
hey @phangs. Your use case is possible. It's easier when using our internal DB but it is possible using an external db too.

**Comment by crudyrudy** [comment]
Especially for mobile devices, a view or edit button on every line is a waste of space. It would be nice to be able to switch to the detail screen in a table by clicking on a row.
